spring-cloud-alicloud-oss/src/main/java/com/alibaba/alicloud/oss/resource/OssStorageResource.java

```diff
@@ -193,4 +216,61 @@ private void assertExisted() throws FileNotFoundException {
 		}
 	}
 
+	public Bucket createBucket() {
+		return this.oss.createBucket(this.bucketName);
+	}
+
+	@Override
+	public boolean isWritable() {
+		return !isBucket() && (this.autoCreateFiles || exists());
+	}
+
+	/**
+     * 获取一个OutputStream用于写操作。
+	 * 注意：写完成后必须关闭该流
+	 * @return
+     * @throws IOException
+	 */
+	@Override
+	public OutputStream getOutputStream() throws IOException {
+		if (isBucket()) {
+			throw new IllegalStateException(
+				"Cannot open an output stream to a bucket: '" + getURI() + "'");
+		}
+		else {
+			OSSObject ossObject;
+
+		    try {
+				ossObject = this.getOSSObject();
+			} catch (OSSException ex) {
+		    	if (ex.getMessage() != null && ex.getMessage().startsWith(MESSAGE_KEY_NOT_EXIST)) {
+		    		ossObject = null;
+				} else {
+		    		throw ex;
+				}
+			}
+
+			if (ossObject == null ) {
+				if (!this.autoCreateFiles) {
+					throw new FileNotFoundException("The object was not found: " + getURI());
+				}
+
+			}
+
+			PipedInputStream in = new PipedInputStream();
+			final PipedOutputStream out = new PipedOutputStream(in);
+
+			executorService.submit(() -> {
```

Comment:
i think it is no need to use ExecutorService.

Reply:
Then this method (getOutputStream()) may not return as the putObject method kept waiting for read()?

Reply:
yes, you are right.

but we construct `ExecutorService ` by each `OssStorageResource`, hope it could by be optimized with Common `ExecutorService`.